tests/isolated_headers/test_isolated_headers.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+import subprocess
+
+from dds_ci.testing import ProjectOpener
+
+
+def test_dependent_src_header_fails(project_opener: ProjectOpener, capfd) -> None:
+    proj = project_opener.open('bad_proj')
+    with pytest.raises(subprocess.CalledProcessError):
+        proj.build()
+    out = capfd.readouterr()
+    errmsg = out.out + '\n' + out.err
+
+    assert 'Syntax check failed' in errmsg
+    assert 'bad_src.hpp' in errmsg
```

Comment:
I'm not sure if what I did is desirable, but I used the error marker and made it include the filename, checking matching a regular expression on the Python side. If error markers like `syntax-check-failed: /path/to/bad-src.hpp`  are undesirable, I can just do a regular error marker; it would be completely fine in this case, since the only way the syntax check can fail is from the single header (assuming the other tests in DDS pass).